docs/api-reference.md

```diff
@@ -34,13 +34,193 @@ it("should add 1 and 1", function()
 end)
 ```
 
-### itFOCUS and itSKIP
+### expect
 ```
-itFOCUS(phrase, callback)
-itSKIP(phrase, callback)
+expect(value: any)
 ```
 
-These methods are special versions of `it` that automatically mark the `it` block as *focused* or *skipped*. They're necessary because `FOCUS` and `SKIP` can't be called inside `it` blocks!
+Creates a new `Expectation`, used for testing the properties of the given value.
+
+Expectations are intended to be read like English assertions. These are all true:
+
+```lua
+-- Equality
+expect(1).to.equal(1)
+expect(1).never.to.equal(2)
+
+-- Approximate equality
+expect(5).to.be.near(5 + 1e-8)
+expect(5).to.be.near(5 - 1e-8)
+expect(math.pi).never.to.be.near(3)
+
+-- Optional limit parameter
+expect(math.pi).to.be.near(3, 0.2)
+
+-- Nil checking
+expect(1).to.be.ok()
+expect(false).to.be.ok()
+expect(nil).never.to.be.ok()
+
+-- Type checking
+expect(1).to.be.a("number")
+expect(newproxy(true)).to.be.a("userdata")
+
+-- Function throwing
+expect(function()
+	error("nope")
+end).to.throw()
+
+expect(function()
+	-- I don't throw!
+end).never.to.throw()
+
+expect(function()
+	error("nope")
+end).to.throw("nope")
+
+expect(function()
+	error("foo")
+end).never.to.throw("bar")
+```
+
+### Lifecycle hooks
+
+The lifecycle hooks are a collection of blocks similar to `it` blocks. They
+can be used to collect setup and teardown code into a common place.
+
+When placed in an `init.spec.lua` file, these hooks will run for all tests
+within that folder.
+
+!!!note
+	The callbacks in the lifecycle hooks are run after all `describe` blocks'
+	callbacks. Be careful not to count on setup or teardown code applying to
+	code not inside an `it` block.
+
+#### afterAll
+```
+afterAll(callback(context: table))
+```
+
+Returns a function after all the tests within its scope run. This is useful if you want to clean up some global state that is used by other tests within its scope.
+
+For example:
+
+```lua
+local DEFAULT_STATE = {
+	hello = "world",
+}
+
+local globalState = DEFAULT_STATE
+
+afterAll(function()
+	globalState = DEFAULT_STATE
+end)
+
+it("SHOULD read globalState", function()
+	expect(globalState.hello).to.equal("world")
+end)
+
+it("SHOULD insert globalState", function()
+	globalState.foo = "bar"
+	expect(globalState.foo).to.equal("bar")
+end)
+```
+
+!!!note
+	If you want to run some cleanup after every test instead of all tests, use `afterEach` instead.
+
+#### afterEach
+```
+afterEach(callback(context: table))
+```
+
+Returns a function after each of the tests within its scope. This is useful if you want to cleanup some temporary state that is created by each test.
+
+```lua
+local DEFAULT_STATE = {
+	hello = "world",
+}
+
+local globalState = DEFAULT_STATE
+
+afterEach(function()
+	globalState = DEFAULT_STATE
+end)
+
+it("SHOULD read globalState", function()
+	expect(globalState.hello).to.equal("world")
+end)
+
+it("SHOULD insert globalState", function()
```

Comment:
If you put the "SHOULD insert" first and have the two tests mess with the same member of globalState, it might better demonstrate the difference between afterEach and afterAll.